src/Components/Browser.JS/src/Services/UriHelper.ts

```diff
@@ -65,8 +65,17 @@ export function navigateTo(uri: string, forceLoad: boolean) {
   const absoluteUri = toAbsoluteUri(uri);
 
   if (!forceLoad && isWithinBaseUriSpace(absoluteUri)) {
+    // It's an internal URL, so do client-side navigation
     performInternalNavigation(absoluteUri, false);
+  } else if (forceLoad && location.href === uri) {
+    // Force-loading the same URL you're already on requires special handling to avoid
+    // triggering browser-specific behavior issues.
+    // For details about what this fixes and why, see https://github.com/aspnet/AspNetCore/pull/10839
+    const temporaryUri = uri + '?';
+    history.replaceState(null, '', temporaryUri);
+    window.location.replace(uri);
```

Comment:
Just to be consistent with the rest of the code.

```suggestion
    location.replace(uri);
```